File: Club/BinarySearch.java
import java.util.Scanner;
import java.io.File;
import java.io.FileNotFoundException;

public class BinarySearch {

     public static void main(String []args) throws FileNotFoundException{
         Scanner sc = new Scanner(new File("texto.txt"));
         int arraySize = sc.nextInt();
         int[] arreglo = new int[arraySize];
         int numQueries = sc.nextInt(); sc.nextLine();
         for(int i = 0; i < arraySize; i++){
             arreglo[i] = sc.nextInt();
         }
         sc.nextLine();
         for(int casos = 0; casos < numQueries; casos++){
           int elemento = sc.nextInt(); sc.nextLine();
           int indice = binarySearch(arreglo, 0, arreglo.length-1, elemento);
           if (indice != -1) {
             while(indice+1 < arreglo.length){
               int nuevaBusqueda = binarySearch(arreglo, indice+1, arreglo.length-1, elemento);
               if (nuevaBusqueda == -1) {
                 break;
               } else {
                 indice = nuevaBusqueda;
               }
             }
           }
           System.out.println(indice);
         }
     }

    private static int binarySearch(int[] lista, int izq, int der, int element){
      if (der < izq) return -1;
        int m = (der + izq) / 2;
        if(lista[m] == element) return m;
        else if (lista[m] > element) return binarySearch(lista, izq, m-1, element);
        else return binarySearch(lista, m+1, der, element);
   }
}

File: Club/BinarySearch.cpp
#include <iostream>
#include <vector>

int binarySearch(std::vector<int>& arreglo, int elemento);
int binarySearchRecursivo(std::vector<int>& arreglo, int izq, int der, int elemento);


int main(){
  std::vector<int> lista = {1, 2, 3, 5, 7};
  std::cout << binarySearch(lista, 5);
  std::cout << binarySearch(lista, 2);
  std::cout << binarySearch(lista, 8);
  return 0;
}

int binarySearch(std::vector<int>& arreglo, int elemento){
  return binarySearchRecursivo(arreglo, 0, arreglo.size()-1, elemento);
}

int binarySearchRecursivo(std::vector<int>& arreglo, int izq, int der, int elemento){
  if (der < izq)
    return -1;
  else {
    int m = (der + izq) / 2;
    if(arreglo[m] == elemento)
      return m;
    else if(arreglo[m] > elemento)
      return binarySearchRecursivo(arreglo, izq, m-1, elemento);
    else
      return binarySearchRecursivo(arreglo, m+1, der, elemento);
    }
}
